### PR TITLE
Fixed ASAN null-dereference write when unable to create thread context memory

### PR DESCRIPTION
**Pull request by nmoinvaz**
ASAN (32-bit) https://oss-fuzz.com/testcase-detail/4589997492666368

### CODE REVIEW

File: blosc/blosc2.c [hunk 1]
```diff
@@ -1667,6 +1672,9 @@ static int do_job(blosc2_context* context) {
       free_thread_context(context->serial_context);
       context->serial_context = create_thread_context(context, 0);
     }
+    if (context->serial_context == NULL) {
+      return BLOSC2_ERROR_THREAD_CREATE;
+    }
```
Comment (FrancescAlted) [inline blosc/blosc2.c:1677]: Couldn't this be just a single `BLOSC_ERROR_NULL` line?

File: blosc/blosc2.c [hunk 2]
```diff
@@ -2623,7 +2631,9 @@ int blosc2_getitem_ctx(blosc2_context* context, const void* src, int32_t srcsize
   if (context->serial_context == NULL) {
     context->serial_context = create_thread_context(context, 0);
   }
-
+  if (context->serial_context == NULL) {
+    return BLOSC2_ERROR_THREAD_CREATE;
+  }
```
Comment (FrancescAlted) [inline blosc/blosc2.c:2636]: Ditto

File: blosc/blosc2.c [hunk 3]
```diff
@@ -2891,7 +2901,9 @@ int init_threadpool(blosc2_context *context) {
     for (tid = 0; tid < context->nthreads; tid++) {
       /* Create a thread context (will destroy when finished) */
       struct thread_context *thread_context = create_thread_context(context, tid);
-
+      if (thread_context == NULL) {
+        return BLOSC2_ERROR_THREAD_CREATE;
+      }
```
Comment (FrancescAlted) [inline blosc/blosc2.c:2906]: Ditto

File: blosc/blosc2.c (hunk 1 above)
Comment (nmoinvaz) [reply blosc/blosc2.c]: Fixed.

File: blosc/blosc2.c (hunk 3 above)
Comment (nmoinvaz) [reply blosc/blosc2.c]: Fixed.

File: blosc/blosc2.c (hunk 2 above)
Comment (nmoinvaz) [reply blosc/blosc2.c]: Fixed.

**Review by FrancescAlted — APPROVED** [review]
LGTM

**Pull request merged**